app/start.sh

```diff
@@ -7,7 +7,9 @@ term_handler() {
 
     # shellcheck source=functions.sh
     source /app/functions.sh
-    remove_all_location_configurations
+    if parse_true "${ACME_HTTP_CHALLENGE_LOCATION:=false}"; then
+        remove_all_location_configurations
+    fi
```

Comment:
I think this cause some intermittent CI failure. We'll tackle this in #1128 or later.
```suggestion
    remove_all_location_configurations
```